contrib/annoy-cmake/CMakeLists.txt

```diff
@@ -1,6 +1,14 @@
 option(ENABLE_ANNOY "Enable Annoy index support" ${ENABLE_LIBRARIES})
 
-if (NOT ENABLE_ANNOY)
+# Annoy index should be disabled with undefined sanitizer. Because of memory storage optimizations
+# (https://github.com/ClickHouse/annoy/blob/9d8a603a4cd252448589e84c9846f94368d5a289/src/annoylib.h#L442-L463)
+# UBSan fails and leads to crash. Simmilar issue is already opened in Annoy repo
+# https://github.com/spotify/annoy/issues/456
+# Problem with aligment can lead to errors like
+# (https://stackoverflow.com/questions/46790550/c-undefined-behavior-strict-aliasing-rule-or-incorrect-alignment)
+# or will lead to crash on arm https://developer.arm.com/documentation/ka003038/latest
+# This issues should be resolved before annoy became non-experimental
```

Comment:
Suggest to add a reference to the cfg setting in braces, e.g. `[...] before annoy can become non-experimental (--> setting "allow_experimental_annoy_index")`